mesheryctl/internal/cli/root/mesh/remove.go

```diff
@@ -2,74 +2,36 @@ package mesh
 
-	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
 	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
-	"github.com/spf13/viper"
 )
 
-var (
-	removeCmd = &cobra.Command{
-		Use:   "remove",
-		Short: "remove a service mesh in the kubernetes cluster",
-		Args:  cobra.MinimumNArgs(0),
-		Long:  `remove service mesh in the connected kubernetes cluster`,
-		Example: `
-// Remove a service mesh
-mesheryctl mesh remove [mesh adapter name]
-		`,
-		PreRunE: func(cmd *cobra.Command, args []string) error {
-			utils.Log.Info("Verifying prerequisites...")
-			mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
-			if err != nil {
-				return errors.Wrap(err, "error processing config")
-			}
-
-			if len(args) < 1 {
-				meshName, err = validateMesh(mctlCfg, "")
-			} else {
-				meshName, err = validateMesh(mctlCfg, args[0])
-			}
-			if err != nil {
-				return errors.Wrap(err, "error validating request")
-			}
-
-			if err = validateAdapter(mctlCfg, meshName); err != nil {
-				return errors.Wrap(err, "adapter not valid")
-			}
-			return nil
-		},
-		RunE: func(cmd *cobra.Command, args []string) error {
-			s := utils.CreateDefaultSpinner(fmt.Sprintf("Removing %s", meshName), fmt.Sprintf("\n%s service mesh removed successfully", meshName))
-			mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
-			if err != nil {
-				return errors.Wrap(err, "error processing config")
-			}
-
-			s.Start()
-			_, err = sendDeployRequest(mctlCfg, meshName, true)
-			if err != nil {
-				return errors.Wrap(err, "error installing service mesh")
-			}
-			s.Stop()
-
-			//log.Infof("Verifying Installation")
-			//s.Start()
-			//_, err = waitForDeployResponse(mctlCfg, meshName)
-			//if err != nil {
-			//	return errors.Wrap(err, "error verifying installation")
-			//}
-			//s.Stop()
-
-			return nil
-		},
-	}
-)
+var removeCmd = &cobra.Command{
+	Use:   "remove [mesh-name]",
+	Short: "Remove a service mesh from the connected Kubernetes cluster",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		s := utils.CreateDefaultSpinner(
+			fmt.Sprintf("Removing %s", meshName),
+			fmt.Sprintf("\n%s removed successfully", meshName),
+		)
+
+		s.Start()
+		_, err = sendOperationRequest(mctlCfg, strings.ToLower(meshName), true)
+		if err != nil {
+			return errors.Wrap(err, "error installing service mesh")
+		}
+		s.Stop()
+
+		return nil
+	},
+}
 
 func init() {
-	removeCmd.Flags().StringVarP(&adapterURL, "adapter", "a", "meshery-istio:10000", "Adapter to use for installation")
-	removeCmd.Flags().StringVarP(&namespace, "namespace", "n", "default", "Kubernetes namespace to be used for deploying the validation tests and sample workload")
-	removeCmd.Flags().StringVarP(&utils.TokenFlag, "token", "t", "", "Path to token for authenticating to Meshery API")
+	removeCmd.Flags().StringVarP(
+		&namespace, "namespace", "n", "default",
+		"Kubernetes namespace where the mesh is deployed",
+	)
```

Comment:
Ok... adapter flag is stripped... no big deal. But I'm more concerned on --token removal.... I'll run and check

Reply:
Ok seems its a global flag now...

Reply:
Yes, `--token` is added in `init()` for the parent `mesh` command in `mesheryctl/internal/cli/root/mesh/mesh.go` (L#79). All of the child commands (`deploy`, `remove`, `validate`) inherit it.

```bash
$ mesheryctl mesh deploy -h
Deploy a service mesh in the connected Kubernetes cluster

Usage:
  mesheryctl mesh deploy [mesh-name] [flags]

Examples:

// Deploy a service mesh interactively:
mesheryctl mesh deploy

// Deploy Linkerd in a specific namespace:
mesheryctl mesh deploy linkerd --namespace linkerd-ns

// Deploy Linkerd and wait for it to be deployed:
mesheryctl mesh deploy linkerd --watch

Flags:
  -h, --help               help for deploy
  -n, --namespace string   Kubernetes namespace where the mesh will be deployed (default "default")
  -w, --watch              Watch for events and verify operation (in beta testing)

Global Flags:
      --config string   path to config file (default "/Users/j.byers/.meshery/config.yaml")
  -t, --token string    Path to token for authenticating to Meshery API
  -v, --verbose         verbose output
```

Reply:
👍👍